lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -891,6 +891,16 @@ void TypeAnalysis::visitOperation(Operation *op,
     return;
   }
 
+  if (isa<AtenTopkOp>(op)) {
```

Comment:
There is currently no support of `AtenTopKOp` in any of the backends of torch-mlir, so this code is not being tested properly.

Reply:
Yes, we have custom backend for `AtenTopKOp`. Is there any suggestion to test it?

Reply:
At the moment there is no way to add this. However, once the [transition to dtype functions](https://github.com/llvm/torch-mlir/issues/1807) is complete, you will be able to add this and test it without needing to have a lowering for the op in torch-mlir

Reply:
how about only add the dtype infer and remove the E2E test？

Reply:
In general, we expect every addition to torch-MLIR to be e2e tested, since this is the only way right now that we can ensure that the behavior of torch-mlir is in accordance with PyTorch's behavior.

Reply:
Thinking about this a bit more, since `AtenTopKOp` does tend to occur many times in a model, you can add it's dtype function into `main`. This will allow you to add tests that ensure your dtype logic agrees with pytorch. See: https://github.com/llvm/torch-mlir/blob/main/docs/adding_abstract_interpretation_functions.md#step-by-step-guide

Reply:
@qingyunqu maybe you could decompose topk into sort+slice?